scipy/optimize/tests/test_linprog.py

```diff
@@ -1955,6 +1991,19 @@ class TestLinprogHiGHSSimplexDual(LinprogHiGHSTests):
     method = "highs-ds"
     options = {}
 
+    def test_lad_regression(self):
+        '''The scaled model should be optimal but unscaled model infeasible.'''
+        c, A_ub, b_ub, bnds = l1_regression_prob()
+        res = linprog(c, A_ub=A_ub, b_ub=b_ub, bounds=bnds,
+                      method=self.method, options=self.options)
+        assert_equal(res.status, 4)
+        assert_('An optimal solution to the scaled '
+                'model was found but' in res.message)
+        assert_(res.x is not None)
+        assert_(np.all(res.slack > -1e-6))
+        assert_(np.all(res.x <= [np.inf if u is None else u for l, u in bnds]))
+        assert_(np.all(res.x >= [-np.inf if l is None else l for l, u in bnds]))
```

Comment:
Apparently plain `assert` is preferred these days. But I'm sure there is plenty of `assert_` in the linprog test suite so this would not be out of place and it's fine with me.

Reply:
Interesting -- I haven't seen anything about this and would be interested to know more.  Is there a thread I can read up on?

The [numpy.testing docs](https://numpy.org/doc/stable/reference/testing.html#testing-guidelines) still recommend `assert_` for this stated reason:

> Within these test methods, assert_() and related functions are used to test whether a certain assumption is valid. If the assertion fails, the test fails. Note that the Python builtin assert should not be used, because it is stripped during compilation with -O.

Reply:
This thread seems relevant: https://github.com/scipy/scipy/issues/8583

Reply:
Right, so `assert_` was recommended, but gh-8583 is probably back when we were using nose. Other maintainers have commented recently that `assert` is preferable with pytest, I think